src/chunk_constraint.c

```diff
@@ -651,17 +652,23 @@ chunk_constraint_delete_metadata(TupleInfo *ti)
 		heap_getattr(ti->tuple, Anum_chunk_constraint_constraint_name, ti->desc, &isnull);
 	int32 chunk_id =
 		DatumGetInt32(heap_getattr(ti->tuple, Anum_chunk_constraint_chunk_id, ti->desc, &isnull));
-	Chunk *chunk = ts_chunk_get_by_id(chunk_id, 0, true);
-	Oid index_relid = get_constraint_index(
-		get_relation_constraint_oid(chunk->table_id, NameStr(*DatumGetName(constrname)), true));
+	/* Get the chunk relid. Note that, at this point, the chunk table can be
+	 * deleted already */
+	Oid chunk_relid = ts_chunk_get_relid(chunk_id, true);
 
-	/*
-	 * If this is an index constraint, we need to cleanup the index
-	 * metadata. Don't drop the index though, since that will happend when
-	 * the constraint is dropped.
-	 */
-	if (OidIsValid(index_relid))
-		ts_chunk_index_delete(chunk, index_relid, false);
+	if (OidIsValid(chunk_relid))
```

Comment:
I believe it cannot be the other case, when it is not valid. I suggest to either remove the condition or replace with assert.

Reply:
It can be invalid, per the comment above.

Basically, there are multiple orders in which events can happen. 

1) Either the chunk and/or constraint is deleted before our own metadata
2) The chunk and/or constraint is deleted after our metadata

This code handles both cases.

Reply:
I understood that the condition is correct, since it specifies `missing_ok = true`, while it was `false` before and will not work correctly with the described case.